sdks/go/pkg/beam/artifact/materialize.go

```diff
@@ -69,41 +74,100 @@ func newMaterialize(ctx context.Context, endpoint string, dependencies []*pipepb
 	return newMaterializeWithClient(ctx, jobpb.NewArtifactRetrievalServiceClient(cc), dependencies, dest)
 }
 
-func newMaterializeWithClient(ctx context.Context, client jobpb.ArtifactRetrievalServiceClient, dependencies []*pipepb.ArtifactInformation, dest string) ([]*jobpb.ArtifactMetadata, error) {
+func newMaterializeWithClient(ctx context.Context, client jobpb.ArtifactRetrievalServiceClient, dependencies []*pipepb.ArtifactInformation, dest string) ([]*pipepb.ArtifactInformation, error) {
 	resolution, err := client.ResolveArtifacts(ctx, &jobpb.ResolveArtifactsRequest{Artifacts: dependencies})
 	if err != nil {
 		return nil, err
 	}
 
-	var md []*jobpb.ArtifactMetadata
+	var artifacts []*pipepb.ArtifactInformation
 	var list []retrievable
 	for _, dep := range resolution.Replacements {
 		path, err := extractStagingToPath(dep)
 		if err != nil {
 			return nil, err
 		}
-		md = append(md, &jobpb.ArtifactMetadata{
-			Name: path,
+		var filePayload pipepb.ArtifactFilePayload
+		if dep.TypeUrn != URNFileArtifact {
+			filePayload = pipepb.ArtifactFilePayload{
+				Path: path,
+			}
+		} else {
+			typePayload := pipepb.ArtifactFilePayload{}
+			if err := proto.Unmarshal(dep.TypePayload, &typePayload); err != nil {
+				return nil, errors.Wrap(err, "failed to parse artifact file payload")
+			}
+			filePayload = pipepb.ArtifactFilePayload{
+				Path:   path,
+				Sha256: typePayload.Sha256,
+			}
+		}
```

Comment:
Perhaps for compactness?
```suggestion
		filePayload := pipepb.ArtifactFilePayload{
				Path: path,
		}
		if dep.TypeUrn == URNFileArtifact {
			typePayload := pipepb.ArtifactFilePayload{}
			if err := proto.Unmarshal(dep.TypePayload, &typePayload); err != nil {
				return nil, errors.Wrap(err, "failed to parse artifact file payload")
			}
			filePayload.Sha256 = typePayload.Sha256
		}
```

If you'd like me go go full readability on you, WRT protos, it's the same either way whether you pre-declare as a pointer type or not. You do not need to dereference to access fields on a pointer to a struct, and escape analysis will determine whether to allocate values to the heap, regardless of if they are pointers or not. If a pointer value never escapes the stack, then it will be stack allocated.

So the following is identical.
```suggestion
		filePayload := &pipepb.ArtifactFilePayload{
				Path: path,
		}
		if dep.TypeUrn == URNFileArtifact {
			typePayload := &pipepb.ArtifactFilePayload{}
			if err := proto.Unmarshal(dep.TypePayload, typePayload); err != nil {
				return nil, errors.Wrap(err, "failed to parse artifact file payload")
			}
			filePayload.Sha256 = typePayload.Sha256
		}
```

(though the remaining &s will need to be removed as appropriate.

Reply:
Done. Thanks for the readability review.